5_evaluate.R

```diff
@@ -47,109 +48,153 @@ p5 <- list(
              },
              format = 'file'),
   
+  # Get vector of evaluation sites, based on matched pred-obs
+  tar_target(p5_eval_sites,
+             p5_nldas_pred_obs %>% pull(site_id) %>% unique()),
+  
+  # Pull out lake depth for each evaluation site
+  tar_target(
+    p5_nldas_depths,
+    purrr::map_df(p5_eval_sites, function(site_id) {
+      site_nml <- p1_nldas_nml_list_subset[[site_id]]
+      tibble(
+        site_id = site_id,
+        lake_depth = site_nml$lake_depth
+      )
+    })
+  ),
+  
   # Prep matched preds for evaluation
   # Add pred_diff column (pred - obs)
   # Set up variables for which bias/accuracy will be calculated
   # Add fields for year, depth_class, doy, doy_bin, season, temp_bin
   tar_target(p5_nldas_pred_obs_eval,
-             prep_data_for_eval(p5_nldas_pred_obs, surface_cutoff_depth = 1, middle_cutoff_depth = 5,
-                                doy_bin_size = 5, temp_bin_size = 2)),
-  
-  # Filter to only surface pred-obs
-  tar_target(p5_nldas_pred_obs_eval_surface,
-             filter(p5_nldas_pred_obs_eval, depth_class=='surface')),
+             prep_data_for_eval(p5_nldas_pred_obs, p5_nldas_depths, surface_max_depth = 1, 
+                                bottom_depth_factor = 0.85, doy_bin_size = 5, temp_bin_size = 2)),
+  
+  # Filter evaluation pred-obs to surface and bottom predictions and set up for group mapping
+  tar_target(p5_nldas_pred_obs_eval_groups,
+             p5_nldas_pred_obs_eval %>%
+               filter(depth_class %in% c('surface','bottom')) %>%
```

Comment:
My read on this is that it is set up to be easy to expand this if we added other depth categories to the output of the above prep target (`p5_nldas_pred_obs_eval` here). So if we added "middle" to that, we'd add `%in% c('surface','middle','bottom')` here. It does make me wonder what other values `depth_class` can have that get filtered out and why the filtering is needed if nothing is getting removed (I can't see the data here, so I'm just speculating).

Reply:
Ignore this, I see the answer below.